.github/workflows/deploy.yml

```diff
@@ -0,0 +1,53 @@
+name: Deploy to Vercel
+
+on:
+  push:
+
+defaults:
+    run:
+        working-directory: ./phone-test
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'yarn'
+          cache-dependency-path: ./phone-test/yarn.lock
+
+      - run: yarn install --frozen-lockfile
+
+      - name: Run tests
+        run: yarn test --watchAll=false
+
+      - name: Cypress run
+        uses: cypress-io/github-action@v6
+        with:
+          start: yarn start
+          build: yarn build
+          wait-on: 'http://localhost:3000' # Waits for above
+          # Records to Cypress Cloud
+          # https://docs.cypress.io/guides/cloud/projects#Set-up-a-project-to-record
+          record: true
+          project: ./phone-test
+
+        env:
+          # For recording and parallelization to work you must set your CYPRESS_RECORD_KEY
+          # in GitHub repo → Settings → Secrets → Actions
+          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
```

Comment:
We need to add CYPRESS_RECORD_KEY to Github Action secrets to make this work. Once secret is added, we can get detailed reports on https://cypress.io